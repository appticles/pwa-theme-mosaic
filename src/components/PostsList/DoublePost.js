import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { postPropType } from '../../views/PostsCarousel/reducer';
import PostCard from '../PostCard/index';

const DoublePost = props => (
  <Grid columns={1} divided>
    <GridColumn className="double-post">
      <GridRow>
        <PostCard post={props.postsList[0]} key={Math.random()} />
      </GridRow>
      <GridRow>
        <PostCard post={props.postsList[1]} key={Math.random()} />
      </GridRow>
    </GridColumn>
  </Grid>
);

DoublePost.propTypes = {
  postsList: PropTypes.arrayOf(postPropType).isRequired,
};

export default DoublePost;
