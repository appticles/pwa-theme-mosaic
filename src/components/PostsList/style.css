/* OVER RIDE semantic UI styles */
/* GRID row */
.double-post .row {
  height: 50%;
}
/* CARD */
.row .ui.card {
  margin-left: auto;
  margin-right: auto;
  padding: 27px;
  border-radius: 0;
}

.row .ui.card .ui.large.image {
  float: none;
  margin: -15px auto 10px;
}
.row .ui.card .ui.large.image img {
  min-width: 63vw;
}
