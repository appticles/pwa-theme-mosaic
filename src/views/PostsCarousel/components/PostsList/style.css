.double-post {
  flex: 1;
  margin-bottom: 45px; }
  .double-post .post-item {
    height: 50%;
    overflow: hidden; }
    .double-post .post-item:first-child {
      border-bottom: 1px solid; }

.single-post {
  flex: 1;
  margin-bottom: 45px; }
  .single-post .post-item {
    height: 90%;
    min-width: 90%;
    margin-top: 24px;
    margin-bottom: 0;
    margin-left: 16px;
    margin-right: 16px;
    border: 1px solid; }
