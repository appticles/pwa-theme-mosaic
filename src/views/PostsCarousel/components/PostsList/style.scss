@import "../../../../utilities/_variables.scss";

.double-post {
  flex: 1;
  margin-bottom: 45px;

  .post-item {
    height: 50%;
    overflow: hidden;

    &:first-child {
      border-bottom: 1px solid;
    }
  }
}

.single-post {
  flex: 1;
  margin-bottom: 45px;

  .post-item {
    height: 90%;
    min-width: 90%;
    margin-top: $base-spacing-md;
    margin-bottom: 0;
    margin-left: $base-spacing;
    margin-right: $base-spacing;
    border: 1px solid;
  }
}
