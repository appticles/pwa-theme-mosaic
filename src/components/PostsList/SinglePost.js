import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import PostCard from '../PostCard/index';
import './style.css';

const SinglePost = props => (
  <Grid>
    <GridColumn>
      <GridRow>
        <PostCard post={props.postsList[0]} key={Math.random()} />
      </GridRow>
    </GridColumn>
  </Grid>
);

SinglePost.propTypes = {
  postsList: PropTypes.array.isRequired,
};
export default SinglePost;
