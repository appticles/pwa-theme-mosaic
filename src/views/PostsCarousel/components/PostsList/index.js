import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridColumn, GridRow, Card } from 'semantic-ui-react';

import { postPropType } from '../../reducer';
import PostCard from '../PostCard';
import './style.css';

const PostsList = (props) => {
  if (props.postsList.length % 2 === 0) {
    return (
      <div>
        <Grid columns={1} divided>
          <GridColumn className="double-post">
            <GridRow>
              <PostCard post={props.postsList[0]} key={props.postsList[0].id} category={props.category} />
            </GridRow>
            <hr />
            <GridRow>
              <PostCard post={props.postsList[1]} key={props.postsList[1].id} category={props.category} />
            </GridRow>
          </GridColumn>
        </Grid>
      </div>
    );
  }

  return (
    <div>
      <Grid className="single-post">
        <GridColumn>
          <GridRow>
            <Card>
              <PostCard post={props.postsList[0]} key={props.postsList[0].id} category={props.category} />
            </Card>
          </GridRow>
        </GridColumn>
      </Grid>
    </div>
  );
};

PostsList.propTypes = {
  postsList: PropTypes.arrayOf(postPropType).isRequired, // TO DO: check rule for eslint
  category: PropTypes.shape({
    params: PropTypes.shape({
      categoryId: PropTypes.string.isRequired,
      categorySlug: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default PostsList;
