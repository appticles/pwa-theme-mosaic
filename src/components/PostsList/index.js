import React, { Component } from 'react';
import PropTypes from 'prop-types';

import SinglePost from './SinglePost';
import DoublePost from './DoublePost';

export default class CategoriesList extends Component {
  getPosts() {
    if (this.props.postsList.length % 2 === 0) {
      return <DoublePost postsList={this.props.postsList} />;
    }
    return <SinglePost postsList={this.props.postsList} />;
  }
  render() {
    return <div>{this.getPosts()}</div>;
  }
}

CategoriesList.propTypes = {
  postsList: PropTypes.array.isRequired,
};
