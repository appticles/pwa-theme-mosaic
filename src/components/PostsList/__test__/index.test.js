import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PostsList from '../index';

describe('Posts list', () => {
  it('should render correctly', () => {
    const mockPosts = [
      {
        id: 1,
        title: {
          rendered: 'Mock title',
        },
        author: 55,
        categories: [43, 54, 23],
        slug: 'mock-slug',
        content: {
          rendered: 'Some mock content to be rendered',
          protected: false,
        },
        excerpt: {
          rendered: 'Some mock exceprt to be rendered',
          protected: true,
        },
        date: 'some mock date',
        _embedded: {
          'wp:featuredmedia': 'someMockImageUrl',
        },
      },
      {
        id: 2,
        title: {
          rendered: 'Mock title',
        },
        author: 55,
        categories: [43, 54, 23],
        slug: 'mock-slug',
        content: {
          rendered: 'Some mock content to be rendered',
          protected: false,
        },
        excerpt: {
          rendered: 'Some mock exceprt to be rendered',
          protected: true,
        },
        date: 'some mock date',
        _embedded: {
          'wp:featuredmedia': 'someMockImageUrl',
        },
      },
    ];
    const output = shallow(<PostsList postsList={mockPosts} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
