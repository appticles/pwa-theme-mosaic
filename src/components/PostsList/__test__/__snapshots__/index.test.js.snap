// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts list should render correctly even number of posts 1`] = `
<div>
  <Grid
    columns={1}
    divided={true}
  >
    <GridColumn
      className="double-post"
    >
      <GridRow>
        <PostCard
          category={
            Object {
              "params": Object {
                "categoryId": "32",
                "categorySlug": "some-slug",
              },
            }
          }
          key="1"
          post={
            Object {
              "_embedded": Object {
                "wp:featuredmedia": "someMockImageUrl",
              },
              "author": 55,
              "categories": Array [
                43,
                54,
                23,
              ],
              "content": Object {
                "protected": false,
                "rendered": "Some mock content to be rendered",
              },
              "date": "some mock date",
              "excerpt": Object {
                "protected": true,
                "rendered": "Some mock exceprt to be rendered",
              },
              "id": 1,
              "slug": "mock-slug",
              "title": Object {
                "rendered": "Mock title",
              },
            }
          }
        />
      </GridRow>
      <hr />
      <GridRow>
        <PostCard
          category={
            Object {
              "params": Object {
                "categoryId": "32",
                "categorySlug": "some-slug",
              },
            }
          }
          key="2"
          post={
            Object {
              "_embedded": Object {
                "wp:featuredmedia": "someMockImageUrl",
              },
              "author": 55,
              "categories": Array [
                43,
                54,
                23,
              ],
              "content": Object {
                "protected": false,
                "rendered": "Some mock content to be rendered",
              },
              "date": "some mock date",
              "excerpt": Object {
                "protected": true,
                "rendered": "Some mock exceprt to be rendered",
              },
              "id": 2,
              "slug": "mock-slug",
              "title": Object {
                "rendered": "Mock title",
              },
            }
          }
        />
      </GridRow>
    </GridColumn>
  </Grid>
</div>
`;

exports[`Posts list should render correctly odd number of posts 1`] = `
<div>
  <Grid>
    <GridColumn>
      <GridRow>
        <Card>
          <PostCard
            category={
              Object {
                "params": Object {
                  "categoryId": "32",
                  "categorySlug": "some-slug",
                },
              }
            }
            key="1"
            post={
              Object {
                "_embedded": Object {
                  "wp:featuredmedia": "someMockImageUrl",
                },
                "author": 55,
                "categories": Array [
                  43,
                  54,
                  23,
                ],
                "content": Object {
                  "protected": false,
                  "rendered": "Some mock content to be rendered",
                },
                "date": "some mock date",
                "excerpt": Object {
                  "protected": true,
                  "rendered": "Some mock exceprt to be rendered",
                },
                "id": 1,
                "slug": "mock-slug",
                "title": Object {
                  "rendered": "Mock title",
                },
              }
            }
          />
        </Card>
      </GridRow>
    </GridColumn>
  </Grid>
</div>
`;
